api/routes/employees.js

```diff
@@ -38,8 +38,26 @@ router.post('/', async (req, res, next) => {
 
 router.get('/statuses', async (req, res, next) => {
     try {
-        const employees = await db.query(queries.getStatuses, { type: db.QueryTypes.SELECT});
-        res.json(employees);
+        const employeesInPostgres = await Employee.findAll({
+            group: ['employee.code', 'employee.fullName', 'employee.active', 'transaction.employeeId', 'transaction.id'],
+            attributes: [
+                'code',
+                'fullName',
+                'active',
+                [sequelize.fn('max', sequelize.col('transaction.originalDate')), 'lastStatusChange']
+            ],
+            include: [{
+                model: Transaction,
+                attributes: ['originalDate', 'employeeId']
+            }]
+        });
+        const employees = employeesInPostgres.map((e) => ({
+            id: e.code,
+            fullName: e.fullName,
+            lastStatusChange: e.lastStatusChange,
+            status: e.active == 1 ? 'in' : 'out'
```

Comment:
e.active indicates if the employee is still working at SFL, we need to make sure that postgres returns only ones that are (so active === true). to check if an employee is in or out we need to check the device id that generated the transaction:

https://github.com/sflpro/status-check-react/blob/master/scheduler/index.js#L42

Reply:
so I need select all actives and add status using deviceId ?

Reply:
yes